packages/typespec-python/src/emitter.ts

```diff
@@ -566,12 +551,11 @@ function emitResponse(
         statusCodes.push(parseInt(response.statusCode));
     }
     return {
-        headers: emitResponseHeaders(context, innerResponse.headers),
+        headers,
         statusCodes: statusCodes,
         addedOn: getAddedOnVersion(context, response.type),
         discriminator: "basic",
         type: type,
-        resultProperty: resultProperty,
```

Comment:
It will cause test failure. Context: https://github.com/Azure/autorest.python/pull/1821/files#r1247454845